src/main/java/dk/kb/util/xml/XpathUtils.java

```diff
@@ -0,0 +1,390 @@
+/* $Id$
+ *
+ * The Summa project.
+ * Copyright (C) 2005-2008  The State and University Library
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package dk.kb.util.xml;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.XMLConstants;
+import javax.xml.namespace.NamespaceContext;
+import javax.xml.namespace.QName;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
+import javax.xml.xpath.XPathExpressionException;
+import javax.xml.xpath.XPathFactory;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Helpers for doing DOM parsing and manipulations. The methods are thread-safe and allows for parallel execution of the
+ * same xpath.
+ */
+public class XpathUtils {
+    private static Logger log = LoggerFactory.getLogger(XpathUtils.class);
+    
+    /**
+     * Importatnt: All access to the xpathCompiler should be synchronized on it since it is not thread safe!
+     */
+    private static final XPath xpathCompiler =
+            XPathFactory.newInstance().newXPath();
+    
+    
+    public static XPathSelector createXPathSelector(String... nsContextStrings) {
```

Comment:
Please provide a unit-test (that also demonstrates how this is used).

Reply:
Done